packages/ui-components/src/FormDialog.tsx

```diff
@@ -0,0 +1,107 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../style/index.css';
+
+import { Dialog } from '@jupyterlab/apputils';
+import { Widget } from '@lumino/widgets';
+
+const DEFAULT_BUTTON_CLASS = 'elyra-DialogDefaultButton';
+/*
+ * Validate required dialog fields upon display
+ * - Provides a generic validation by checking if required form fields are populated
+ * - Expect required fields in dialog body to contain attribute: data-form-required
+ * - NOTE: The Dialog widget will skip any validation upon pressing 'enter' key and resolve the dialog as is
+ *
+ * @params
+ *
+ * options - The dialog setup options
+ * formValidationFunction - Optional validation function
+ *
+ * returns a call to dialog display
+ */
+export const showFormDialog = async (
+  options: Partial<Dialog.IOptions<any>>,
+  formValidationFunction: any
+): Promise<Dialog.IResult<any>> => {
+  const dialogBody = options.body;
+  const dialog = new Dialog(options);
+
+  if (formValidationFunction && typeof formValidationFunction === 'function') {
+    formValidationFunction(dialog);
+  } else {
+    if (typeof dialogBody !== 'string') {
+      const requiredFields = (dialogBody as Widget).node.querySelectorAll(
+        '[data-form-required]'
+      );
+
+      if (requiredFields && requiredFields.length > 0) {
+        // Keep track of all fields already validated. Start with an empty set.
+        let fieldsValidated = new Set();
+
+        const defaultButtonIndex =
+          options.defaultButton || options.buttons.length - 1;
+        const defaultButton = dialog.node
+          .querySelector('.jp-Dialog-footer')
+          .getElementsByTagName('button')[defaultButtonIndex];
+
+        disableDialogButton(defaultButton);
+
+        requiredFields.forEach((element: any) => {
+          // First deal with the case the field has already been pre-populated
+          element.value
+            ? fieldsValidated.add(element)
+            : fieldsValidated.delete(element);
+
+          const fieldType = element.tagName.toLowerCase();
+
+          if (fieldType === 'select') {
+            element.addEventListener('change', (event: any) => {
+              // Update fieldsValidated set accordingly
+              event.target.value
+                ? fieldsValidated.add(element)
+                : fieldsValidated.delete(element);
+
+              // Only enable defaultButton if all required fields are validated
+              fieldsValidated.size === requiredFields.length
+                ? enableDialogButton(defaultButton)
+                : disableDialogButton(defaultButton);
```

Comment:
same here. since this code block is being called a couple times, i think it was good the way you had it before as its own function